services/src/main/java/org/openlearn/service/StorageService.java

```diff
@@ -51,12 +76,13 @@ public StorageService(final FileRepository fileRepository,
 	 * @return the persisted entity
 	 */
 	//TODO cbernal fix blue documentation
-	public FileInformation store(final MultipartFile file, Long assignmentId, Long portfolioId) {
+	public FileInformationDTO store(final MultipartFile file, Long assignmentId, Long portfolioId) {
```

Comment:
Perhaps. Taylor didn't make it transactional initially, but I don't know his reasoning. My only guess is that since storing a file *can* take some time (depending on the size), it might be because we didn't want to hog resources with a long running transaction.

Reply:
I'd have to look into how H2 implements transactions to have any clear indications of what tradeoffs there might be.